modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/configuration/persistence/listener/BaseConfigurationModelListener.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.segments.context.vocabulary.internal.configuration.persistence.listener;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.configuration.persistence.listener.ConfigurationModelListener;
+import com.liferay.portal.configuration.persistence.listener.ConfigurationModelListenerException;
+import com.liferay.portal.kernel.util.LocaleThreadLocal;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Dictionary;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.ResourceBundle;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Janis Zhang
+ */
+public class BaseConfigurationModelListener
+	implements ConfigurationModelListener {
+
+	@Override
+	public void onBeforeSave(String pid, Dictionary<String, Object> properties)
+		throws ConfigurationModelListenerException {
+
+		String entityFieldName = String.valueOf(
+			properties.get("entityFieldName"));
+
+		if (Validator.isNull(entityFieldName)) {
+			throw new ConfigurationModelListenerException(
+				ResourceBundleUtil.getString(
+					_getResourceBundle(),
+					"please-enter-a-valid-session-property-name"),
+				_clazz, getClass(), properties);
+		}
+
+		if (_isDefined(
+				String.valueOf(properties.get("assetVocabularyName")),
+				String.valueOf(properties.get("companyId")), entityFieldName)) {
+
+			throw new DuplicatedSegmentsContextVocabularyConfigurationModelListenerException(
+				ResourceBundleUtil.getString(
+					_getResourceBundle(),
+					"this-field-is-already-linked-to-one-vocabulary"),
+				_clazz, getClass(), properties);
+		}
+	}
+
+	protected void init(ConfigurationAdmin configurationAdmin, Class<?> clazz) {
+		_configurationAdmin = configurationAdmin;
+		_clazz = clazz;
```

Comment:
@JanisZhang Please remove this init method, and declare an abstract method `getConfigurationClass`, and implement it in sub classes.